pkg/nuctl/command/patch.go

```diff
@@ -0,0 +1,366 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/dashboard/resource"
+	"github.com/nuclio/nuclio/pkg/errgroup"
+	"github.com/nuclio/nuclio/pkg/platform"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/nuclio/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	SessionEndpoint   = "api/sessions"
+	FunctionsEndpoint = "nuclio-proxy/api/functions"
+)
+
+type patchCommandeer struct {
+	cmd             *cobra.Command
+	rootCommandeer  *RootCommandeer
+	patchOptionsMap map[string]string
+	patchOptions    *resource.PatchOptions
+	httpClient      *http.Client
+}
+
+func newPatchCommandeer(ctx context.Context, rootCommandeer *RootCommandeer) *patchCommandeer {
+	commandeer := &patchCommandeer{
+		rootCommandeer: rootCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "patch",
+		Short: "Apply a patch to a resource",
+	}
+
+	cmd.PersistentFlags().StringToStringVarP(&commandeer.patchOptionsMap,
+		"patch-options",
+		"o",
+		map[string]string{},
+		"Patch options, as a key=value (e.g. -o key1=value1). Can be used multiple times.")
+
+	cmd.AddCommand(
+		newPatchFunctionsCommandeer(ctx, commandeer).cmd,
+	)
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (c *patchCommandeer) initialize() error {
+
+	// initialize http client
+	c.httpClient = &http.Client{
```

Comment:
must set default timeout (60s) and remove `InsecureSkipVerify` from default, get that as a flag